### PR TITLE
8.0 project partner name

### PR DESCRIPTION
**Pull request by franksongfeng**
add _get_partner_ref(self, proj)  for  the class AnalyticAccount

### CODE REVIEW

**Comment by franksongfeng** [comment]
@seb-elico @noahzaozao @elicoidal @victormartinelicocorp @xuan139 pls review

File: project_partner_name/models/analytic_account_id.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            full_names = self._get_full_names(elmt, 6)
+
+            if isinstance(full_names, list) and full_names:
+        
+                project_ids = self.pool['project.project'].search(
+                    cr, uid,
+                    [('analytic_account_id', '=', id)],
+                    context=context
+                )
+
```
Comment (noahzaozao) [inline project_partner_name/models/analytic_account_id.py]: too much lines , remove this line

File: project_partner_name/models/analytic_account_id.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                        project_ids[0],
+                        context=context
+                    )
+
+                    partner_ref = self._get_partner_ref(project_obj)
+
+                    if partner_ref:
+                        full_names[0] = "%s - %s" % (partner_ref, full_names[0])
+
+
```
Comment (noahzaozao) [inline project_partner_name/models/analytic_account_id.py]: too much line , remove this line

**Review by noahzaozao — CHANGES_REQUESTED** [review]

**Review by noahzaozao — APPROVED** [review]

File: project_partner_name/models/analytic_account_id.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                        project_ids[0],
+                        context=context
+                    )
+                    partner_ref = self._get_partner_ref(project_obj)
+                    if partner_ref:
+                        full_names[0] = "%s - %s" % (partner_ref, full_names[0])
+                full_names.reverse()  # order on First Root Last Leaf                
+            res.append((id, ' / '.join(full_names)))
+
+        return res
```
Comment (noahzaozao) [inline project_partner_name/models/analytic_account_id.py]: Please add a line at end of the py file

**Comment by elicoidal** [comment]
@franksongfeng please attend pending details.
other than that :+1:  (needs testing)

@seb-elico @noahzaozao @victormartinelicocorp @xuan139 pls review

**Comment by noahzaozao** [comment]
LGTM

**Comment by xuan139** [comment]
LGTM

**Pull request merged**